runtime/executor/src/main/java/org/apache/nemo/runtime/executor/datatransfer/RunTimeMessageOutputCollector.java

```diff
@@ -46,36 +46,46 @@
   private final IRVertex irVertex;
   private final PersistentConnectionToMasterMap connectionToMasterMap;
   private final TaskExecutor taskExecutor;
+  private final ControlMessage.RunTimePassType runTimePassType;
 
   public RunTimeMessageOutputCollector(final String taskId,
                                        final IRVertex irVertex,
                                        final PersistentConnectionToMasterMap connectionToMasterMap,
-                                       final TaskExecutor taskExecutor) {
+                                       final TaskExecutor taskExecutor,
+                                       final ControlMessage.RunTimePassType runTimePassType) {
     this.taskId = taskId;
     this.irVertex = irVertex;
     this.connectionToMasterMap = connectionToMasterMap;
     this.taskExecutor = taskExecutor;
+    this.runTimePassType = runTimePassType;
   }
 
   @Override
   public void emit(final O output) {
-    final Map<Object, Long> aggregatedMessage = (Map<Object, Long>) output;
     final List<ControlMessage.RunTimePassMessageEntry> entries = new ArrayList<>();
-    aggregatedMessage.forEach((key, size) ->
-      entries.add(
-        ControlMessage.RunTimePassMessageEntry.newBuilder()
-          // TODO #325: Add (de)serialization for non-string key types in data metric collection
-          .setKey(key == null ? NULL_KEY : String.valueOf(key))
-          .setValue(size)
-          .build())
-    );
-
+    switch (runTimePassType) {
```

Comment:
Can you remove this switch?
It'd be better to keep the runtime agnostic to different types of runtime passes.
This switch seems to suggest that one needs to modify the runtime for every new run-time optimization pass, which is probably not what we want.